src/models/AppState.js

```diff
@@ -126,24 +126,26 @@ function (Okta, Q, Factor, BrowserFeatures, Errors) {
       if (this.settings.get('features.securityImage')) {
         var self = this;
         this.listenTo(this, 'change:username', function (model, username) {
-          getSecurityImage(this.get('baseUrl'), username, this.get('deviceFingerprint'))
-            .then(function (image) {
-              model.set('securityImage', image.securityImage);
-              model.set('securityImageDescription', image.securityImageDescription);
-              model.unset('deviceFingerprint'); //Fingerprint can only be used once
-            })
-            .fail(function (jqXhr) {
-            // Only notify the consumer on a CORS error
-              if (BrowserFeatures.corsIsNotEnabled(jqXhr)) {
-                self.settings.callGlobalError(new Errors.UnsupportedBrowserError(
-                  Okta.loc('error.enabled.cors')
-                ));
-              }
-              else {
-                throw jqXhr;
-              }
-            })
-            .done();
+          if (username) {
```

Comment:
when `username` is empty, it will not actually make API call https://github.com/okta/okta-signin-widget/blob/master/src/models/AppState.js#L47-L51

what's exact problem here?

Reply:
I guess the problem is at https://github.com/okta/okta-signin-widget/blob/master/src/models/AppState.js#L43